pkgs/development/libraries/audiofile/default.nix

```diff
@@ -26,7 +27,25 @@ stdenv.mkDerivation rec {
   };
 
   # fix build with gcc9
-  NIX_CFLAGS_LINK = lib.optional (stdenv.system == "i686-linux") "-lgcc";
+  NIX_CFLAGS_LINK =
+    lib.optional (stdenv.system == "i686-linux") "-lgcc"
+    ++ lib.optional static "-lstdc++";
+
+  # Even when statically linking, libstdc++.la is put in dependency_libs here,
+  # and hence libstdc++.so passed to the linker, just pass -lstdc++ (in
+  # NIX_CFLAGS_LINK) and let the compiler do what it does best.
+  # (libaudiofile.la is a generated file, so we have to run `make` that far
+  # first).
+  #
+  # Without this, the executables in this package (sfcommands and examples)
+  # fail to build: https://github.com/NixOS/nixpkgs/issues/103215
+  #
+  # There might be a more sensible way to do this with autotools, but I am not
+  # smart enough to discover it.
+  preBuild = lib.optionalString static ''
+    make -C libaudiofile $makeFlags
+    sed -i "s/dependency_libs=.*/dependency_libs='''/" libaudiofile/libaudiofile.la
+  '';
```

Comment:
You don't need this stuff anymore, right? Also, this could go to master then.

Reply:
Sadly this does need to be in there, otherwise the linker tries to link libstdc++.so (as specified by libstdc++.la)